### PR TITLE
feat!: Merging RC branch to main

### PR DESCRIPTION
**Pull request by sbansla**
# Fixes #

A short description of what this PR does.
Merging 6.0.0-rc branch to main with Breaking change
Changes made as per [doc](https://docs.google.com/document/d/1Ncn-P-0eINdxFErVNGT9AfAGfwBQUF4OQzZX5RL3d5Q/edit)

 - [x] PR is marked as feat! (Breaking Change)
 - [x] Readme: Removed RC installation steps
 - [x] Upgrade: Added steps to upgrade to 6.x.x from 5.x.x
 - [x] Contributing: Edited heading "Found an issue" to identify oai generated files
 - [x] Changes: Removed RC branch changes
 - [x] Removed test/Twilio.Test/Rest package and included src/Twilio/Rest package from rc branch.
 - [ ] Included cluster test from rc branch

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-csharp/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/test-and-deploy.yml [hunk 1]
```diff
@@ -28,12 +28,27 @@ jobs:
 
       - name: Build & Test
         run: make test
+      
+      - name: Run Cluster Test
```
Comment (childish-sambino) [inline .github/workflows/test-and-deploy.yml:32]: If sonar coverage also runs cluster tests, is this step needed?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,12 +30,19 @@ If you find a bug in the source code or a mistake in the documentation, you can
 help us by submitting [an issue][issue-link]. If the file in which the error
 exists has this header:
 ```
-"""
-This code was generated by
-\ / _    _  _|   _  _
- | (_)\/(_)(_|\/| |(/_  v1.0.0
-      /       /
-"""
```
[the remark this answers is not in the document]
Comment (sbansla) [reply CONTRIBUTING.md:38]: addressed

File: .github/workflows/test-and-deploy.yml (hunk 1 above)
Comment (sbansla) [reply .github/workflows/test-and-deploy.yml]: addressed

File: .github/workflows/test-and-deploy.yml [hunk 3]
```diff
@@ -26,14 +26,16 @@ jobs:
         with:
           dotnet-version: '3.1.x'
 
-      - name: Build & Test
```
Comment (childish-sambino) [inline .github/workflows/test-and-deploy.yml:29]: This needs to stay so that we run unit tests for forks.

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**